app/controllers/api/v1/impersonations_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class ImpersonationsController < API::V1::APIController
+      skip_before_action :authenticate_user!
+      skip_after_action :verify_authorized
+
+      def create
+        user, token = Impersonation::Authenticator.new(params[:auth]).authenticate!
+
+        response.headers.merge!(
+          user.build_auth_headers(token.token, token.client)
+        ).merge!(
+          Impersonation::Headers.new(user).build_impersonation_header
+        )
```

Comment:
```suggestion
        response.headers.merge!(
          user.build_auth_headers(token.token, token.client),
          Impersonation::Headers.new(user).build_impersonation_header
        )
```

I think this is the same

Reply:
@santib I tried this but it didn't work